docs/grpc-compatibility.md

```diff
@@ -0,0 +1,98 @@
+# gRPC compatibility
```

Comment:
Put some thought into whether this is actually the right title when there isn't compatibility, but it seemed still fine but open to any thoughts